frame/staking/src/lib.rs

```diff
@@ -960,6 +960,56 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// The list of thresholds separating the various voter bags.
+		///
+		/// Voters are separated into unsorted bags according to their vote weight. This specifies
+		/// the thresholds separating the bags. A voter's bag is the largest bag for which the
+		/// voter's weight is less than or equal to its upper threshold.
+		///
+		/// When voters are iterated, higher bags are iterated completely before lower bags. This
+		/// means that iteration is _semi-sorted_: voters of higher weight tend to come before
+		/// voters of lower weight, but peer voters within a particular bag are sorted in insertion
+		/// order.
+		///
+		/// # Expressing the constant
+		///
+		/// This constant must be sorted in strictly increasing order. Duplicate items are not
```

Comment:
both of these can and should be checked in an `integrity_test`

Reply:
https://crates.parity.io/frame_support/traits/trait.Hooks.html#method.integrity_test

Reply:
Yes, and in fact they are, [here](https://github.com/paritytech/substrate/blob/89db32eef8b1dcfa5fbdb448a0de6f9b8b2970e5/frame/staking/src/lib.rs#L1564-L1567).